exporter/awsxrayexporter/translator/segment.go

```diff
@@ -338,14 +359,73 @@ func makeXRayAttributes(attributes map[string]string, indexedAttrs []string, ind
 				defaultMetadata[key] = value
 			}
 		}
-		if len(defaultMetadata) > 0 {
-			metadata["default"] = defaultMetadata
+
+		if storeResource {
+			resource.Attributes().ForEach(func(key string, value pdata.AttributeValue) {
+				key = "otel.resource." + key
+				annoVal := annotationValue(value)
+				_, indexed := indexedKeys[key]
+				if annoVal != nil && indexed {
+					key = fixAnnotationKey(key)
+					annotations[key] = annoVal
+				} else {
```

Comment:
Not knowing the true context of this functionality, is the metadata fallback still desired if this isn't an indexed key?

Reply:
I'm leaning towards data is always better than no data - a log message seems like it could be useful though. I don't know if I've seen any log_once pattern elsewhere, log_every would be very spammy though. Any recommended pattern for log_once? Sorry for bothering

Reply:
I don't believe there is an established mechanism for this in the collector at the moment, but it would be a nice feature if a limited logger were accessible apart from the standard service one.  The docs mainly say not to log extensively: https://github.com/open-telemetry/opentelemetry-collector/blob/master/CONTRIBUTING.md#logging

Briefly looking it seems zap [provides a way](https://github.com/uber-go/zap/blob/master/zapcore/sampler.go#L142) to do this, but I'm not sure it would ideally be part of these changes: 

```
sampledLogger := zap.New(zapcore.NewSamplerWithOptions(logger.Core(), <duration>, <first>, <thereafter>))
```

Reply:
Thanks, let me leave this as is for now and add some sampled logging or similar if there's a better pattern, or someone asks for it.